cmake/modules/FindSanitizers.cmake

```diff
@@ -34,6 +34,7 @@ if(Sanitizers_address_COMPILE_OPTIONS OR Sanitizers_leak_COMPILE_OPTIONS)
   # ASAN_LIBRARY will be read by ceph.in to preload the asan library
   find_library(ASAN_LIBRARY
     NAMES
+      libasan.so.8
```

Comment:
I wonder if there is a better way to do this though.  This started with 3, 4 and 5, then 6 was added for GCC 11 and now 8 is being added for GCC 12...

It looks like the original motivation was to prefer the latest "known" version in cases where `libasan.so` is older (such as when a new compiler is installed on an old LTS distro)?

Reply:
there is this repo: https://github.com/arsenm/sanitizers-cmake
but could not see how they find the correct libasan.

Reply:
I think it happens here: https://github.com/arsenm/sanitizers-cmake/blob/master/cmake/asan-wrapper.  Instead of attempting to "find" `libasan.so` explicitly, they just parse `ldd` output.

In our case, `ASAN_LIBRARY` variable is used to preload libasan into `ceph` tool (a Python program) so it's not directly applicable.  But anywhere where `ceph` tool is present, `rados` tool (a C++ program) should be present as well, so perhaps we could do what `asan-wrapper` does on that?

Reply:
the wrapper is for runtime. i thought that if we don't specify `ASAN_LIBRARY` we fail at link time?

Reply:
> the wrapper is for runtime. i thought that if we don't specify ASAN_LIBRARY we fail at link time?

i don't think we explicitly link to asan; when i set `WITH_ASAN=ON WITH_UBSAN=ON`, it adds these flags to both the compiler and linker lines: `-fsanitize=address;-fsanitize=undefined;-fno-sanitize=vptr;-fno-omit-frame-pointer`

the toolchain seems to be adding that library dependency itself, so `ldd` does seem like the best way to identify which

> In our case, ASAN_LIBRARY variable is used to preload libasan into ceph tool (a Python program) so it's not directly applicable. But anywhere where ceph tool is present, rados tool (a C++ program) should be present as well, so perhaps we could do what asan-wrapper does on that?

maybe we could put that logic in vstart_environment.sh instead? then the python commands wouldn't have to mess with it